scripts/wsmux.sh

```diff
@@ -36,14 +39,81 @@ wsmux() {
 	while (( $# > 0 )); do
 		case $1 in
 			-h|--help)
-				echo "Usage: wsmux [OPTION]... [CATKIN_WORKSPACE]"
+				echo "Usage: wsmux [OPTION]... [WORKSPACE]"
 				echo "Quick catkin workspace switcher."
 				echo ""
 				echo "Option		GNU long option		Meaning"
+				echo "-b [WORKSPACE]	--bind			Bind an SSH client to a workspace"
+				echo "-c		--connect		Select the SSH client's workspace"
 				echo "-h		--help			Display the help menu"
-				echo "-l		--list			List the detected catkin workspaces"
-				echo "-s		--show			Show the selected catkin workspace"
+				echo "-l		--list			List the detected workspaces"
+				echo "-s		--show			Show the selected workspace"
+				SELECTION="false"
+				shift 1
+				;;
+			-b|--bind)
+
+				# Ensure that a client has connected via SSH
+				if [[ -z "$SSH_CLIENT" ]]; then
+					echo "This is not an SSH session, so there is no client to bind"
+
+				# Prompt the user to check themself if the workspace is not valid
+				elif [[ ! -f ~/$2/devel/setup.sh ]]; then
+					echo "The specified workspace is not valid. I don't trust like that..."
+					echo "Please specify the name of a workspace in the home directory"
+
+				else
+					source $WS_CONFIG_FILE
+
+					# Replace any binding that already exists for this SSH client
+					local REPLACED="false"
+					for (( BINDING_INDEX=0; BINDING_INDEX < ${#WORKSPACE_BINDINGS[@]}; BINDING_INDEX++ )); do
+						if [[ "$(echo ${WORKSPACE_BINDINGS[BINDING_INDEX]} | cut -d ':' -f1)" == "$(echo $SSH_CLIENT | cut -d ' ' -f1)" ]]; then
+							WORKSPACE_BINDINGS[BINDING_INDEX]="$(echo $SSH_CLIENT | cut -d ' ' -f1):$2"
```

Comment:
Your DHCP leases are all handled based on the MAC addresses you gave me, so the IPs will not change.